bundles/org.openhab.binding.mielecloud/src/main/java/org/openhab/binding/mielecloud/internal/handler/MieleBridgeHandler.java

```diff
@@ -153,6 +158,39 @@ public void handleRemoval() {
         super.handleRemoval();
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * Note: This is the same implementation as in {@link BaseThingHandler} except that it is not checked whether the
```

Comment:
Hm, this does not really make sense to me. This method is again one that is called by the framework and should behave accordingly. If the framework calls it and nothing has changed about the configuration, no re-initialization is necessary and should hence not be done.

Isn't what you want to do to introduce a new private method instead, which you can call from `reconfigureBridge()` and that re-initializes what is necessary (without calling `initialize()` nor `dispose()`)?

Reply:
> Hm, this does not really make sense to me. This method is again one that is called by the framework and should behave accordingly. If the framework calls it and nothing has changed about the configuration, no re-initialization is necessary and should hence not be done.

The problem we try to solve by doing this is that the configuration of the bridge is made up of the configuration parameters and the current access token. We could check whether the access token changed prior to calling `dispose()` and `initialize()` here. Would that be a valid solution?

> Isn't what you want to do to introduce a new private method instead, which you can call from reconfigureBridge() and that re-initializes what is necessary (without calling initialize() nor dispose())?

I could do so, but it is necessary to do everything that is done in `initialize()` and `dispose()`. I could put all of that into separate methods, like `internalInitialize()` and `internalDispose()` and call these from `reconfigureBridge()`, `initialize()` and `dispose()` but I don't see the advantage of doing this apart from not calling framework methods. It would rather confuse me when I would be reading such code.

Reply:
Maybe you better see the advantage if you think of better names for the private methods, e.g. something like `startBackgroundThreads()` and `stopBackgroundThreads()` or something describing what the code does.
My problem with `dispose()` is that the name of the method signals that after calling it, the object is/can be garbage collected and that it is not in use anymore. Calling such a method and still continuing to work with the object seems wrong. 
Likewise, `initialize()` as a method is similar to the OSGi services `activate()`, which is called exactly once in the lifecycle of an object. So any additional call seems fishy.
These issues can be escaped by defining your own private methods with a suitable naming.

Reply:
I did as you requested and decided to use `initializeWebservice` and `disposeWebservice` as the webservice component used by the handler is disposed and initialized again. Hope this is fine now.